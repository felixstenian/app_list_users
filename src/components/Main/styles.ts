import styled from 'styled-components'

export const Wrapper = styled.main`
  width: 100%;
  background-color: ${({ theme }) => theme.colors.white};
`
