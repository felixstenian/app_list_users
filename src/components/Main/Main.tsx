import { FiSearch } from 'react-icons/fi'

import { Header, Row, Text, TextField } from 'components'

import * as Style from './styles'

const Main = () => {
  const list = [
    {
      id: 1,
      avatar: 'https://randomuser.me/api/portraits/women/2.jpg',
      name: 'Genésia',
      nac: 'Brazil',
      email: 'genesia.barbosa@example.com'
    },
    {
      id: 2,
      avatar: 'https://randomuser.me/api/portraits/women/2.jpg',
      name: 'Genésia',
      nac: 'Brazil',
      email: 'genesia.barbosa@example.com'
    },
    {
      id: 3,
      avatar: 'https://randomuser.me/api/portraits/women/2.jpg',
      name: 'Genésia',
      nac: 'Brazil',
      email: 'genesia.barbosa@example.com'
    }
  ]
  return (
    <Style.Wrapper>
      <Header>LOGO</Header>

      <Row
        mt={20}
        width="100%"
        maxWidth="100rem"
        justifyContent="space-around"
        alignItems="center"
        m="5rem auto"
        // bg={['secondary', 'primary', 'mainBg']}
        flexDirection={['column', 'column', 'row']}
      >
        <Row>
          <TextField placeholder="Pesquisar" icon={<FiSearch />} />
        </Row>

        <Row alignItems="center">
          <Text>Filtrar por:</Text>
          <Text variant="caption">Nacionalidade</Text>
          <select />
          <Text variant="caption">Genero</Text>
          <select />
        </Row>
      </Row>

      <Row width="100%" maxWidth="100rem" m="10rem auto" flexDirection="column">
        {list.map((user) => (
          <Row
            key={user.id}
            bg="#fafa"
            width="100%"
            justifyContent="space-between"
            alignItems="center"
            p={10}
            borderRadius="8px"
            my="5px"
            flexDirection={['column', 'column', 'row']}
          >
            <Row alignItems="center">
              <figure>
                <img
                  width="50%"
                  style={{
                    borderRadius: '50px'
                  }}
                  src={`${user.avatar}`}
                  alt="alt"
                />
              </figure>
              <div>{user.name}</div>
            </Row>
            <div>{user.nac}</div>
            <div>{user.email}</div>
          </Row>
        ))}
      </Row>
    </Style.Wrapper>
  )
}

export default Main
