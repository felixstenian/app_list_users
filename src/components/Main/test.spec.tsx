import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils'
import Main from '.'

describe('<Main />', () => {
  it('1. should render the reading', () => {
    const { container } = renderWithTheme(<Main />)

    expect(screen.getByRole('button', { name: /buscar/i })).toBeInTheDocument()
    expect(screen.getByText(/LOGO/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the color correctly', () => {
    const { container } = renderWithTheme(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#FAFAFA'
    })
  })
})
