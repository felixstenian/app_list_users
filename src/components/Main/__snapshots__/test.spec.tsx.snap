// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> 1. should render the reading 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 20px;
  margin: 5rem auto;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  max-width: 100rem;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 10rem auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  max-width: 100rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  margin-top: 5px;
  margin-bottom: 5px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background-color: #fafa;
  border-radius: 8px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  height: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #06092B;
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c8 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c9 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4rem;
  border: 0.2rem solid;
  background: #EAEAEA;
  padding: 0.8rem;
  border-color: #EAEAEA;
}

.c4:focus-within {
  box-shadow: 0 0 0.5rem #8F8F8F;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #8F8F8F;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c5 > svg {
  width: 2.2rem;
  height: 100%;
}

.c6 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-right: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: calc(100% - 2.2rem);
}

.c6:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c6:-webkit-autofill::first-line {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c0 {
  width: 100%;
  background-color: #FAFAFA;
}

@media screen and (min-width:40em) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c11 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:52em) {
  .c11 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c11 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:52em) {
  .c11 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<main
  class="c0"
>
  <header
    class="c1"
  >
    LOGO
  </header>
  <div
    class="c2"
    width="100%"
  >
    <div
      class="c3"
    >
      <div
        class=""
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <line
                x1="21"
                x2="16.65"
                y1="21"
                y2="16.65"
              />
            </svg>
          </div>
          <input
            class="c6"
            placeholder="Pesquisar"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="c7"
    >
      <body
        class="c8"
      >
        Filtrar por:
      </body>
      <caption
        class="c9"
      >
        Nacionalidade
      </caption>
      <select />
      <caption
        class="c9"
      >
        Genero
      </caption>
      <select />
    </div>
  </div>
  <div
    class="c10"
    width="100%"
  >
    <div
      class="c11"
      width="100%"
    >
      <div
        class="c7"
      >
        <figure>
          <img
            alt="alt"
            src="https://randomuser.me/api/portraits/women/2.jpg"
            style="border-radius: 50px;"
            width="50%"
          />
        </figure>
        <div>
          Genésia
        </div>
      </div>
      <div>
        Brazil
      </div>
      <div>
        genesia.barbosa@example.com
      </div>
    </div>
    <div
      class="c11"
      width="100%"
    >
      <div
        class="c7"
      >
        <figure>
          <img
            alt="alt"
            src="https://randomuser.me/api/portraits/women/2.jpg"
            style="border-radius: 50px;"
            width="50%"
          />
        </figure>
        <div>
          Genésia
        </div>
      </div>
      <div>
        Brazil
      </div>
      <div>
        genesia.barbosa@example.com
      </div>
    </div>
    <div
      class="c11"
      width="100%"
    >
      <div
        class="c7"
      >
        <figure>
          <img
            alt="alt"
            src="https://randomuser.me/api/portraits/women/2.jpg"
            style="border-radius: 50px;"
            width="50%"
          />
        </figure>
        <div>
          Genésia
        </div>
      </div>
      <div>
        Brazil
      </div>
      <div>
        genesia.barbosa@example.com
      </div>
    </div>
  </div>
</main>
`;
